plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestJdbcCachingConnectorSmokeTest.java

```diff
@@ -34,9 +40,29 @@ protected QueryRunner createQueryRunner()
                 .put("metadata.cache-missing", "true")
                 .put("allow-drop-table", "true")
                 .build();
+        this.h2SqlExecutor = new JdbcSqlExecutor(properties.get("connection-url"), new Properties());
         return createH2QueryRunner(REQUIRED_TPCH_TABLES, properties);
     }
 
+    @Test
+    public void testFlushJdbcMetadataCacheProcedureCallable()
```

Comment:
Add another test to test the remote identifier cache flush.

Reply:
I was trying to add it. But identifier mapping is only caching schema and table names.
Also cache key is based on name it self which causes it to resolve when new un-existing value is asked.
So we just resolve it again. 
I tried it to check on errors like. Caching table name, renaming it outside of Trino, and making query to it (we should get error from target DB), and then asking again for un-existing table so we should get error from Trino as cache would be all-ready update. But errors are the same :( so I can't recognise if cache was flushed or not. 
Any suggestions, how we could approach it ?

Reply:
> Also cache is implement to reload when value is not found.

See `metadata.cache-missing`. Enabling it caches cache misses too.

Reply:
Thx will try it out.

Reply:
@hashhar `metadata.cache-missing` is implemented only for metadata cache in JdbcClient.
Then one implemented in CachingIdentifierMapping is not using such behaviour.
Also maybe I'm missing something but i think this step:
> Then you can create the missing table using H2 (use mixed case to ensure remote name cache gets used).

will not fill remote names cache (CachingIdentifierMapping) as we will make it no H2 omitting Trino and it's caches.

Reply:
The test isn't working because TestingH2JdbcModule doesn't install the IdentifierMappingModule and is using the no-cache DefaultIdentifierMapping.

I also notice now that the module also doesn't bind a CachingJdbcClient. Only a plain TestingH2JdbcClient is created.

Reply:
You can verify this by trying to query a lowercase table in H2.

e.g. In H2 `CREATE TABLE "lowercase"` and then you won't be able to query it from Trino.

I think in the end a test like below should pass.

```java
    @Test
    public void testFlushIdentifierCacheProcedureFlushCache()
    {
        assertUpdate("CREATE TABLE cached_name AS SELECT * FROM nation", 25);
        String query = "SELECT name, regionkey FROM cached_name";
        assertQuerySucceeds(query);

        // Rename to lowercase name outside Trino (H2 stores unquoted names as uppercase)
        h2SqlExecutor.execute("ALTER TABLE tpch.cached_name RENAME TO tpch.\"cached_name\"");

        // Should fail as Trino has old identifier cached
        assertThatThrownBy(() -> getQueryRunner().execute(query))
                .hasMessageMatching("(?s)Table \"CACHED_NAME\" not found.*");

        // Should succeed after flushing Trino cache
        getQueryRunner().execute(getSession(), "CALL system.flush_metadata_cache()");
        assertQuerySucceeds(query);
    }
```

Reply:
From beginning I'm trying to implement test in https://github.com/trinodb/trino/blob/master/plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseCaseInsensitiveMappingTest.java as there are test covering test scenarios with IdentifierMapping . 
Let me try doing as you suggest:
- wired IdentifierMappingModule in TestJdbcCachingConnectorSmokeTest
- add test as above

Reply:
@hashhar thank you for your help - much appreciated.
This is the trick i was missing `Rename to lowercase name outside Trino (H2 stores unquoted names as uppercase)`  
To keeps tests together I've modified TestingH2JdbcClient to use IdentifierMapping from Guice context instead of always default and added test together in TestJdbcCachingConnectorSmokeTest